x/posts/types/models/post.go

```diff
@@ -106,13 +108,33 @@ type Post struct {
 	OptionalData   OptionalData   `json:"optional_data,omitempty" yaml:"optional_data,omitempty"` // Arbitrary data that can be used from the developers
 	Creator        sdk.AccAddress `json:"creator" yaml:"creator"`                                 // Creator of the Post
 	Attachments    Attachments    `json:"attachments,omitempty" yaml:"attachments,omitempty"`     // Contains all the attachments that are shared with the post
-	PollData       *PollData      `json:"poll_data,omitempty" yaml:"poll_data"`                   // Contains the poll details, if existing
+	PollData       *PollData      `json:"poll_data,omitempty" yaml:"poll_data,omitempty"`         // Contains the poll details, if existing
 }
 
-func NewPost(id, parentID PostID, message string, allowsComments bool, subspace string,
+// computeID computes a post ID based on the content of the given post.
+// nolint
+func computeID(parentID PostID, message, subspace string, allowsComments bool, creationTime time.Time, creator sdk.AccAddress,
+	attachments Attachments, pollData *PollData) PostID {
+	id := parentID.String() + message + subspace + strconv.FormatBool(allowsComments) + creationTime.String() +
```

Comment:
Instead of creating a string concatenating the different parts of a post, I think that it might be better to use the JSON serialization. This would also make it easier for extenal libraries to implement this method on their own and compute the IDs locally as well. 

Ideally, the same approach as when signing a `Message` should be followed: 

1. Serialize the entire `Post` as a JSON. 
2. Order the inner keys alphabetically in ascending order. 
3. Take the result and hash it using SHA-256. 

This would be ideal as it would always produce the same ID not matter the order of the serialized keys. What do you think about this?

Reply:
Yup. This should be good enough for third parties.
Also the two first points are done at the same time inside `json.Marshal` so no big deal in using such approach. 👍